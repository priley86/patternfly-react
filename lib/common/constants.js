'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var TOAST_NOTIFICATION_TYPES = exports.TOAST_NOTIFICATION_TYPES = ['danger', 'error', 'warning', 'success', 'info'];

var ALERT_TYPES = exports.ALERT_TYPES = ['danger', 'error', 'warning', 'success', 'info'];