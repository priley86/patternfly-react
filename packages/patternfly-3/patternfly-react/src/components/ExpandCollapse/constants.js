export const ALIGN_LEFT = 'left';
export const ALIGN_CENTER = 'center';

export const ALIGN_TYPES = [ALIGN_LEFT, ALIGN_CENTER];
