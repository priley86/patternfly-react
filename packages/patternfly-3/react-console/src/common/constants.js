export const NONE_TYPE = '_none_';
export const SERIAL_CONSOLE_TYPE = 'SerialConsole';
export const VNC_CONSOLE_TYPE = 'VncConsole';
