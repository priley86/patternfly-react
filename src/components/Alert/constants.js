export const ALERT_TYPES = ['danger', 'error', 'warning', 'success', 'info'];
