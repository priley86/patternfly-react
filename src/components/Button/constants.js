export const BUTTON_BS_STYLES = ['danger', 'default', 'primary', 'link'];
