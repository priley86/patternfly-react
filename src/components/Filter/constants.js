export const FILTER_SELECT_TYPE = 'select';
export const FILTER_COMPLEX_SELECT_TYPE = 'complex-select';

export const FILTER_VALUE_SELECTED = 'filter-value';
export const FILTER_CATEGORY_SELECTEDY = 'filter-category';
