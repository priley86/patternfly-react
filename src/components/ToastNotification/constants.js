export const TOAST_NOTIFICATION_TYPES = [
  'danger',
  'error',
  'warning',
  'success',
  'info',
];
