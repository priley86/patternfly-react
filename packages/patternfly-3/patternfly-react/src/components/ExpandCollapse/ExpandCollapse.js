import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Button } from '../Button';

import { ALIGN_LEFT, ALIGN_CENTER, ALIGN_TYPES } from './constants';

class ExpandCollapse extends React.Component {
  state = { expanded: false };

  onClick = () => {
    this.setState(prevState => ({ expanded: !prevState.expanded }));
  };

  render() {
    const { children, textCollapsed, textExpanded, align, className } = this.props;

    const alignClass = classNames({
      'expand-collapse-pf-left': align === ALIGN_LEFT,
      'expand-collapse-pf-center': align === ALIGN_CENTER
    });

    return (
      <div className={classNames(['expand-collapse-pf', className])}>
        <div className={alignClass}>
          {this.state.expanded ? <span className="fa fa-angle-down" /> : <span className="fa fa-angle-right" />}
          <Button bsStyle="link" onClick={this.onClick}>
            {this.state.expanded ? textExpanded : textCollapsed}
          </Button>
        </div>
        {this.state.expanded && children}
      </div>
    );
  }
}

ExpandCollapse.propTypes = {
  children: PropTypes.any.isRequired,

  className: PropTypes.string /** Top-level custom class */,
  textCollapsed: PropTypes.string /** Text for the link in collapsed state */,
  textExpanded: PropTypes.string /** Text for the link in expanded state */,
  align: PropTypes.oneOf(ALIGN_TYPES) /** Align the link to the left or center. Default: left. */
};

ExpandCollapse.defaultProps = {
  className: '',
  textCollapsed: 'Show Advanced Options',
  textExpanded: 'Hide Advanced Options',
  align: ALIGN_LEFT
};

export default ExpandCollapse;
