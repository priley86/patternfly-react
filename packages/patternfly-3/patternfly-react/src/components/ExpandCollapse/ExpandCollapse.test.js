import React from 'react';
import { mount, render } from 'enzyme';

import ExpandCollapse from './ExpandCollapse';
import { ALIGN_LEFT, ALIGN_CENTER } from './constants';

test('ExpandCollapse with content', () => {
  const view = mount(
    <ExpandCollapse>
      <div id="content">My text</div>
    </ExpandCollapse>
  );

  expect(view.find('#content')).toHaveLength(0);
  expect(view.find('span.fa-angle-right')).toHaveLength(1);
  expect(view.find('span.fa-angle-down')).toHaveLength(0);
  expect(view.find('.btn-link').text()).toEqual('Show Advanced Options');

  const button = view.find('.btn-link');
  expect(button).toHaveLength(1);
  button.simulate('click');
  expect(view.find('#content')).toHaveLength(1);
  expect(view.find('span.fa-angle-right')).toHaveLength(0);
  expect(view.find('span.fa-angle-down')).toHaveLength(1);
  expect(view.find('.btn-link').text()).toEqual('Hide Advanced Options');
  button.simulate('click');
  expect(view.find('#content')).toHaveLength(0);
  expect(view.find('span.fa-angle-right')).toHaveLength(1);
  expect(view.find('span.fa-angle-down')).toHaveLength(0);
  expect(view.find('.btn-link').text()).toEqual('Show Advanced Options');
});

test('localized ExpandCollapse', () => {
  const view = mount(
    <ExpandCollapse textCollapsed="Click to expand" textExpanded="Click to collapse">
      <div id="content">My text</div>
    </ExpandCollapse>
  );
  expect(view).toMatchSnapshot();
  view.find('.btn-link').simulate('click');
  expect(view).toMatchSnapshot();
});

test('aligned ExpandCollapse', () => {
  const def = render(
    <ExpandCollapse>
      <div id="content">My text</div>
    </ExpandCollapse>
  );
  expect(def.find('.expand-collapse-pf-left')).toHaveLength(1);
  expect(def.find('.expand-collapse-pf-center')).toHaveLength(0);

  const left = render(
    <ExpandCollapse align={ALIGN_LEFT}>
      <div id="content">My text</div>
    </ExpandCollapse>
  );
  expect(left.find('.expand-collapse-pf-left')).toHaveLength(1);
  expect(left.find('.expand-collapse-pf-center')).toHaveLength(0);

  const center = render(
    <ExpandCollapse align={ALIGN_CENTER}>
      <div id="content">My text</div>
    </ExpandCollapse>
  );
  expect(center.find('.expand-collapse-pf-left')).toHaveLength(0);
  expect(center.find('.expand-collapse-pf-center')).toHaveLength(1);
});
