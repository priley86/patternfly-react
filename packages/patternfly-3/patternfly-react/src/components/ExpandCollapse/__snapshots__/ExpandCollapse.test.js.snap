// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`localized ExpandCollapse 1`] = `
<ExpandCollapse
  align="left"
  className=""
  textCollapsed="Click to expand"
  textExpanded="Click to collapse"
>
  <div
    className="expand-collapse-pf"
  >
    <div
      className="expand-collapse-pf-left"
    >
      <span
        className="fa fa-angle-right"
      />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="link"
        disabled={false}
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Click to expand
        </button>
      </Button>
    </div>
  </div>
</ExpandCollapse>
`;

exports[`localized ExpandCollapse 2`] = `
<ExpandCollapse
  align="left"
  className=""
  textCollapsed="Click to expand"
  textExpanded="Click to collapse"
>
  <div
    className="expand-collapse-pf"
  >
    <div
      className="expand-collapse-pf-left"
    >
      <span
        className="fa fa-angle-down"
      />
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="link"
        disabled={false}
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Click to collapse
        </button>
      </Button>
    </div>
    <div
      id="content"
    >
      My text
    </div>
  </div>
</ExpandCollapse>
`;
